include/nbl/builtin/hlsl/bxdf/ndf/beckmann.hlsl

```diff
@@ -63,34 +83,22 @@ NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
 struct BeckmannCommon<T,false NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
 {
     using scalar_type = T;
-    using dg1_query_type = SBeckmannDG1Query<scalar_type>;
-    using g2g1_query_type = SBeckmannG2overG1Query<scalar_type>;
 
     template<class MicrofacetCache NBL_FUNC_REQUIRES(ReadableIsotropicMicrofacetCache<MicrofacetCache>)
     scalar_type D(NBL_CONST_REF_ARG(MicrofacetCache) cache)
     {
         scalar_type NdotH2 = cache.getNdotH2();
         scalar_type nom = exp2<scalar_type>((NdotH2 - scalar_type(1.0)) / (log<scalar_type>(2.0) * a2 * NdotH2));
         scalar_type denom = a2 * NdotH2 * NdotH2;
-        return numbers::inv_pi<scalar_type> * nom / denom;
+        return hlsl::mix(bit_cast<scalar_type>(numeric_limits<scalar_type>::infinity), numbers::inv_pi<scalar_type> * nom / denom, a2 > numeric_limits<scalar_type>::min);
```

Comment:
if-statement is fine to use here and exit early, so you don't end up evaluating all the stuff above